libtiledbsoma/src/reindexer/reindexer.h

```diff
@@ -74,24 +75,16 @@ class IntIndexer {
         lookup(keys.data(), results.data(), keys.size());
     }
     IntIndexer(){};
-    /**
-     * Constructor with the same arguments as map_locations
-     */
-    IntIndexer(const int64_t* keys, int size, int threads);
-    IntIndexer(const std::vector<int64_t>& keys, int threads)
-        : IntIndexer(keys.data(), keys.size(), threads) {
-    }
+    IntIndexer(std::shared_ptr<tiledbsoma::SOMAContext> context);
     virtual ~IntIndexer();
 
    private:
     /*
      * The created 64bit hash table
      */
     kh_m64_s* hash_;
-    /*
-     * TileDB threadpool
-     */
-    std::shared_ptr<tiledbsoma::ThreadPool> tiledb_thread_pool_ = nullptr;
+
+    std::shared_ptr<ThreadPool> thread_pool_ = nullptr;
```

Comment:
Probably a question for @nguyenv ... should resources like the IntIndexer hold a reference to the ThreadPool, or to the Context?  I had assumed that when the Context is destructed, it would also tear down the thread pool.  I don't _think_ the above would cause problems as it uses a shared_ptr, but wanted to confirm...

Reply:
Actually it was originally her suggestion that IntIndex holds the reference to the threadpool.

Reply:
Great - ty for clarifying!

Reply:
There might have been some miscommunication because I don't think the `ThreadPool` should reside in here but in the `SOMAContext`.